pages/book/types.mdx

```diff
@@ -1,43 +1,50 @@
-# Tact type system
+# Type system overview
 
-Every variable, item, and value in a Tact program has a type:
+import { Callout } from 'nextra/components'
 
-* Primitives: Int, Bool, Slice, Cell, Builder, String and StringBuilder;
-* Map
-* Structs and Messages
-* Contracts and Traits
+Every variable, item, and value in Tact programs has a type. They can be:
 
-Also, most primitive types, structs, and messages could be nullable.
+* One of the [primitive types](#primitive-types)
+* [Maps](#maps)
+* Composite types, such as [Structs and Messages](#structs-and-messages)
+* or [Contracts](#contracts) and [Traits](#traits)
+
+Also, many of those types [can be made nullable](/book/defining-types#optionals).
 
 ## Primitive types
 
-* Int - all integers in Tact are 257-bit signed integers.
-* Bool - classical boolean with true/false values.
-* Address - standard address.
-* Slice, Cell, Builder - low-level primitive of TON VM.
-* String - type that represents text strings in TON VM.
-* StringBuilder - helper type that allows you to concatenate strings in a gas-efficient way
+* Int — all numbers in Tact are 257-bit signed integers, but [smaller representations](/book/integers#serialization) can be used to reduce storage costs.
+* Bool — classical boolean with `true` and `false` values.
+* Address — standard [smart contract address](https://docs.ton.org/learn/overviews/addresses#address-of-smart-contract) in TON Blockchain.
+* Slice, Cell, Builder — low-level primitives of TON VM.
+* String — represents text strings in TON VM.
+* StringBuilder — helper type that allows you to concatenate strings in a gas-efficient way.
+* `null` — represents the intentional absence of any other value, makes the variables into [optionals](/book/defining-types#optionals).
```

Comment:
From the user perspective `null` is not a type, it only exists at the implementation level. One cannot write `let foo : null = null` or have something like `get fun foo() : null { ... }` or have a field of type `null`, etc.

Reply:
Yup, I'll clarify this

Reply:
I'd just remove `null` from the list of types

Reply:
Agreed